ttnn/cpp/ttnn/operations/normalization.hpp

```diff
@@ -0,0 +1,106 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "tt_eager/tt_dnn/op_library/layernorm/layernorm_op.hpp"
+
+namespace ttnn {
+namespace operations {
+namespace normalization {
+
+
+inline ttnn::Tensor layer_norm(
+    const ttnn::Tensor& input_tensor,
+    float epsilon,
+    std::optional<const ttnn::Tensor> & weight,
+    std::optional<const ttnn::Tensor>& bias,
+    std::optional<const ttnn::Tensor>& residual_input_tensor,
+    const MemoryConfig& memory_config,
+    //std::optional<const LayerNormShardedMultiCoreProgramConfig>& program_config
+    std::optional<const LayerNormProgramConfig>& program_config
+) {
+
+    const LayerNormProgramConfig&  actual_program_config = program_config.value_or(LayerNormDefaultProgramConfig{}); // = .has_value() ? program_config : LayerNormDefaultProgramConfig{};
+    /*if (!program_config.has_value()) {
+        actual_program_config = LayerNormDefaultProgramConfig{};
+    }
+    else {
+        actual_program_config = program_config;
+    }*/
+
+    if (residual_input_tensor.has_value()) {
+        return tt::operations::primary::add_layernorm(input_tensor, residual_input_tensor.value(), epsilon, weight, bias, memory_config, actual_program_config);
+    }
+    else {
+        return tt::operations::primary::layernorm(input_tensor, epsilon, weight, bias, memory_config, actual_program_config);
+    }
+    /*
+    if program_config is None:
+        if residual_input_tensor is not None:
+            output_tensor = ttnn.experimental.tensor.add_layernorm(
+                input_tensor, residual_input_tensor, epsilon, weight, bias, output_mem_config=memory_config
+            )
+        else:
+            output_tensor = ttnn.experimental.tensor.layernorm(
+                input_tensor,
+                epsilon,
+                weight,
+                bias,
+                output_mem_config=memory_config,
+            )
+    else:
+        if residual_input_tensor is not None:
+            output_tensor = ttnn.experimental.operations.primary.add_layernorm(
+                input_tensor,
+                residual_input_tensor,
+                epsilon,
+                weight,
+                bias,
+                output_mem_config=memory_config,
+                program_config=program_config,
+            )
+        else:
+            output_tensor = ttnn.experimental.operations.primary.layernorm(
+                input_tensor, epsilon, weight, bias, output_mem_config=memory_config, program_config=program_config
+            )
+
+    return output_tensor*/
+}
+
+}
+}
+}
+
+#if 0
```

Comment:
What is the intent here?

Reply:
Extra code that is now removed. Branch was still WIP. Should have made that clear